### DISCUSSION TITLE
邮件通知触发两次？

### DISCUSSION THREAD

**Discussion by xiamuguizhi** [ArtalkJS/Artalk#239] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/19650297/184080014-6c6ad04e-be1c-479f-a39c-0018f41c3c11.png)

别人评论通知会有两封邮件，一封正常，一封收件人错误！ 基本前后1秒延迟！！ 三个人这样了

![image](https://user-images.githubusercontent.com/19650297/184080116-f2c0a6c5-c590-46dd-89a2-5ac9db05fbc6.png)

真的奇怪了，只要是有评论邮件通知  管理员 就会这样  一封邮件 收件人 admin <admin@example.com>  一封正常！

![image](https://user-images.githubusercontent.com/19650297/184361263-cf65d835-96a5-4d32-bdaa-95058d9d3d12.png)

**Comment by xiamuguizhi** [comment]
基本每次有人回复就会触发 这个问题..

**Comment by qwqcode** [comment]
可以检查一下配置文件 `artalk-go.yml` 的 `admin_users` 配置项中，是否存在名为 `admin` 邮箱为 `admin@example.com` 的管理员账号。

参考：https://artalk.js.org/guide/backend/email.html

> “当用户「新增评论」或「回复评论」时，ArtalkGo 会向「普通用户」或「管理员」发送邮件。”

参考：https://artalk.js.org/guide/backend/multi-site.html#%E8%B6%85%E7%BA%A7%E7%AE%A1%E7%90%86%E5%91%98%E3%80%81%E6%99%AE%E9%80%9A%E7%AE%A1%E7%90%86%E5%91%98

> “当页面有新的评论时，邮件会发送给该站点的全部管理员，但你可以配置 receive_email 强制禁用它。”

**Comment by xiamuguizhi** [comment]
@qwqcode  你说的我都看了好几次没发现问题。

![image](https://user-images.githubusercontent.com/19650297/184373949-47c5c0a2-5d07-4cc4-a454-b9356a268970.png)

反着就是   访客 评论 -> 邮件通知 （管理员） -> 发送邮件通知 （正常）-> 同时又发送了一封到 （admin <admin@example.com>）

![image](https://user-images.githubusercontent.com/19650297/184375796-bb8c365b-2114-4632-adbe-107761cd11d0.png)


而且我看了一下 admin@example.com  是默认配置的 我的配置文件 是没有这个的 ，那是否 是那些机制导致的？

版本 ： Artalk (v2.1.10/dfba394)

**Reply by qwqcode** [reply, marked as answer]
可以看看数据库里面的 users 表中，是否有 `name` 为 `admin` 的用户

我猜测可能是之前默认配置启动程序的时候，创建了一个 `admin` 用户，导致这个管理员一直都存在

```sql
SELECT * FROM `users` WHERE name =  'admin'
```

**Comment by xiamuguizhi** [comment]
@xiamuguizhi 刚洗澡完 ，把数据库下载看了一下还真是！！

![image](https://user-images.githubusercontent.com/19650297/184385294-931f47fc-8252-4149-be17-d7e3bdb373a7.png)

我直接在数据库删除了，除了ID不连续了以外应该没问题了吧！

**Reply by qwqcode** [reply]
是的，应该没问题了，不过会有缓存依然还在，你可以重启一下 ArtalkGo 的 docker 容器

**Reply by qwqcode** [reply]
这是一个逻辑上的缺陷，感谢发现这个问题

**Reply by xiamuguizhi** [reply]
> 是的，应该没问题了，不过会有缓存依然还在，你可以重启一下 ArtalkGo 的 docker 容器

现在一切正常，没有出现前面提到的情况了。
